posts/2021-01-22-jwtbuilder-redirects-21001.adoc

```diff
@@ -0,0 +1,182 @@
+
+To solve the problems presented, Open Liberty 21.0.0.1 ahs introduced a new `redirecttorelativeurl` web container property that will tell the application whether or not to construct absoulte URLs from relative redirect links. This property can be set in the `server.xml` file:
+
+[source, xml]
+----
+<server>
+  ...
+    <webContainer redirecttorelativeurl="true"/>
+  ...
+</server>
+----
+
+For more information:
+
+* link:https://openliberty.io/docs/21.0.0.1/reference/feature/servlet-4.0.html[Open Liberty Servlet Documentation]
+
+[#conf]
+=== MicroProfile Config 2.0 enters general availability
+
+link:https://github.com/eclipse/microprofile-config[MicroProfile Config] enables application configuration from different locations (ConfigSources). MicroProfile Config 2.0 in Open Liberty consumes link:https://github.com/smallrye/smallrye-config[SmallRye's implementation] of the specification.
+
+MicroProfile Config 2.0 is a major version release (from 1.4) which introduces Functional and API/SPI changes in Microprofile Config 2.0.
+
+For a more in-depth look at the new capabilities offered by MicroProfile 2.0, take a look at the full explanation in the 20.0.0.12-beta blog post, link:https://openliberty.io/blog/2020/10/27/microprofile40-jakartaee9-beta-200012.html#conf[here].
+
+
+To enable MicroProfile Config 2.0 in your app, add the following to your `server.xml`:
+
+[source, xml]
+----
+<featureManager>
+   <feature>mpConfig-2.0</feature>
+</featureManager>
+----
+
+[#bugs]
+== Notable bugs fixed in this release
+
+We’ve spent some time fixing link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A200011+label%3A%22release+bug%22[bugs in 21.0.0.1], including the following issues:
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/14936[Improved recovery when core components are reinstalled at runtime]
++
+Previously, Open Liberty contained a flaw in the detection of changes to the JARs that compose the server implementation. When such changes were detected Liberty would force the JAR to be uninstalled and installed again. In most cases this allowed the runtime to recover and function properly. In specific cases, where some core component was re-installed, Liberty would not properly recover and it would result in some Java packages to be unavailable for class loading.  For example, on Java 11 the package `javax.xml.soap` would become unavailable to the application class loaders. 
++
+The most common environment where this occurred was running Open Liberty in OpenShift. This behavior has now been corrected for Open Liberty 21.0.0.1.
```

Comment:
link "Open Liberty in Openshift" to https://openliberty.io/docs/latest/deployment-openshift.html

Reply:
Added hyperlink.